src/main/java/com/guysfromusa/carsgame/game_state/dtos/GameState.java

```diff
@@ -1,24 +1,54 @@
 package com.guysfromusa.carsgame.game_state.dtos;
 
+import com.guysfromusa.carsgame.control.Message;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
 
+@Slf4j
 public class GameState {
 
-    private Map<String, List<Movement>> carsMovementMap = new HashMap<>();
+    @Getter
+    private final String gameName;
+
+    @Getter
+    private final BlockingQueue<Message> movementsQueue = new LinkedBlockingQueue<>();
+
+    @Setter @Getter
+    private volatile boolean roundInProgress = false;
+
+    private Map<String, List<Movement>> movementsHistoryByCar = new HashMap<>();
+
+    public GameState(String gameName) {
+        this.gameName = gameName;
+    }
+
+    public CompletableFuture<String> addMovementToExecute(Message message) {
```

Comment:
Comparable as above - generic name

Reply:
probably we should switch to 'commands' move/undo/addCar/turn